### ISSUE TITLE
A11y reports missing href on a tag with role="button"

### ISSUE THREAD

**Issue by cloudlena** [sveltejs/kit#6158]
### Describe the bug

Having an `a` tag with `role="button"`, I get the A11y warning `Warn: A11y: <a> element should have an href attribute`. To me, that doesn't make sense because if the `a` tag acts as a button, it shouldn't have an `href` attribute.

I cannot make the `a` tag into a proper `button` tag because I'm using https://bulma.io/ and that would mess up the styling of the element.

 I think, if an `a` tag has a `role="button"` attribute, the warning should not be raised (or even inverted to send a warning if there is a `role="button"` attribute and at the same time an `href`).

### Reproduction

Create ` <a role="button>My Button</a>` element in a `.svelte` file.

### System Info

```shell
System:
    OS: Linux 5.19 Arch Linux
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 9.08 GB / 15.33 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 18.7.0 - /usr/bin/node
    npm: 8.18.0 - /usr/bin/npm
  Browsers:
    Brave Browser: 104.1.42.97
    Chromium: 104.0.5112.101
    Firefox: 103.0.2
  npmPackages:
    @sveltejs/adapter-static: next => 1.0.0-next.39
    @sveltejs/kit: next => 1.0.0-next.428
    svelte: ^3.44.0 => 3.49.0
    vite: ^3.0.0 => 3.0.9
```

### Severity

annoyance

**Comment by Conduitry** [comment]
This came up before in https://github.com/sveltejs/svelte/issues/1719 - `<a role=button>` is bad because browsers won't tab to it. If you really need this, there are ways of disabling the warning - but this is something we do want to warn about.
